aws/resource_aws_macie_s3_bucket_association_test.go

```diff
@@ -0,0 +1,174 @@
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSMacieS3BucketAssociationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSMacieS3BucketAssociationConfig_basic(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSMacieS3BucketAssociationExists("aws_macie_s3_bucket_association.test"),
+					resource.TestCheckResourceAttr("aws_macie_s3_bucket_association.test", "classification_type.0.one_time", "false"),
+				),
+			},
+			{
+				Config: testAccAWSMacieS3BucketAssociationConfig_basicOneTime(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSMacieS3BucketAssociationExists("aws_macie_s3_bucket_association.test"),
+					resource.TestCheckResourceAttr("aws_macie_s3_bucket_association.test", "classification_type.0.one_time", "true"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSMacieS3BucketAssociation_accountIdAndPrefix(t *testing.T) {
+	rInt := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSMacieS3BucketAssociationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSMacieS3BucketAssociationConfig_accountIdAndPrefix(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSMacieS3BucketAssociationExists("aws_macie_s3_bucket_association.test"),
+					resource.TestCheckResourceAttr("aws_macie_s3_bucket_association.test", "classification_type.0.one_time", "false"),
+				),
+			},
+			{
+				Config: testAccAWSMacieS3BucketAssociationConfig_accountIdAndPrefixOneTime(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSMacieS3BucketAssociationExists("aws_macie_s3_bucket_association.test"),
+					resource.TestCheckResourceAttr("aws_macie_s3_bucket_association.test", "classification_type.0.one_time", "true"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAWSMacieS3BucketAssociationDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).macieconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_macie_s3_bucket_association" {
+			continue
+		}
+
+		req := &macie.ListS3ResourcesInput{}
+		acctId := rs.Primary.Attributes["member_account_id"]
+		if acctId != "" {
+			req.MemberAccountId = aws.String(acctId)
+		}
+
+		for {
+			resp, err := conn.ListS3Resources(req)
+			if err != nil {
+				return err
+			}
+
+			for _, v := range resp.S3Resources {
+				if aws.StringValue(v.BucketName) == rs.Primary.Attributes["bucket_name"] && aws.StringValue(v.Prefix) == rs.Primary.Attributes["prefix"] {
+					return fmt.Errorf("S3 resource %s/%s is not dissociated from Macie", rs.Primary.Attributes["bucket_name"], rs.Primary.Attributes["prefix"])
+				}
+			}
+
+			if resp.NextToken == nil {
+				break
+			}
+			req.NextToken = resp.NextToken
+		}
+	}
+	return nil
+}
+
+func testAccCheckAWSMacieS3BucketAssociationExists(name string) resource.TestCheckFunc {
```

Comment:
This function should perform similar logic as the destroy check above to ensure the S3 resources are appropriately allocated in Macie.

Reply:
👍

Reply:
👍